File: scripts/prepare_vercel_public.py
"""Copy built frontend assets into the Vercel static output directory."""

from __future__ import annotations

from pathlib import Path
import shutil
import sys

PROJECT_ROOT = Path(__file__).resolve().parents[1]
DIST_DIR = PROJECT_ROOT / "ui" / "dist"
PUBLIC_DIR = PROJECT_ROOT / "public"


def main() -> int:
    """Copy built UI assets into the static output directory."""
    if not DIST_DIR.exists():
        print(f"Missing built UI directory: {DIST_DIR}", file=sys.stderr)
        return 1

    shutil.rmtree(PUBLIC_DIR, ignore_errors=True)
    PUBLIC_DIR.mkdir(parents=True, exist_ok=True)

    for child in DIST_DIR.iterdir():
        if child.name == "index.html":
            continue
        target = PUBLIC_DIR / child.name
        if child.is_dir():
            shutil.copytree(child, target)
            continue
        shutil.copy2(child, target)

    print(f"Prepared Vercel public assets in {PUBLIC_DIR}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: scripts/prepare-vercel-public.mjs
import { cp, mkdir, readdir, rm } from "node:fs/promises";
import path from "node:path";

const projectRoot = process.cwd();
const outDir = path.join(projectRoot, "ui", "out");
const publicDir = path.join(projectRoot, "public");

async function main() {
	await rm(publicDir, { recursive: true, force: true });
	await mkdir(publicDir, { recursive: true });

	for (const child of await readdir(outDir, { withFileTypes: true })) {
		const sourcePath = path.join(outDir, child.name);
		const targetPath = path.join(publicDir, child.name);
		await cp(sourcePath, targetPath, { recursive: true });
	}

	console.log(`Prepared Vercel public assets in ${publicDir}`);
}

main().catch((error) => {
	console.error(error instanceof Error ? error.message : error);
	process.exitCode = 1;
});
